syntax = "proto3";
package pt.ulisboa.tecnico.nameserver.contract;

option java_multiple_files = true;

message ServerAddress {
  string host = 1;
  int32 port = 2;
}

message ServerEntry  {
  ServerAddress address = 1;
  string qualifier = 2;
}

message RegisterRequest {
  string serviceName = 1;
  ServerAddress address = 2;
  string qualifier = 3;
}

message RegisterResponse {}

message LookUpRequest {
  string serviceName = 1;
  string qualifier = 2;
}

message LookUpResponse {
  repeated ServerEntry  ServerEntry = 1;
}

message DeleteRequest {
  string serviceName = 1;
  ServerAddress address = 2;
}

message DeleteResponse {}

service NamingServerService {
  rpc register(RegisterRequest) returns (RegisterResponse) {}
  rpc lookup(LookUpRequest) returns (LookUpResponse) {}
  rpc delete(DeleteRequest) returns (DeleteResponse) {}
}